### ISSUE TITLE
My Suggestion how to contribute to improve upon the compile feature

### ISSUE THREAD

**Issue by franzhaas** [construct/construct#1106]
I would like to contribute to the compile feature, and I would like to suggest to go about it like this.:

First, I took some time to understand the core.py module better. And my understanding of the compile feature reached a useful level. During this time I made a lot of experiments, the ideas listed here have shown usefulness during these experiments.

# First steps
1.) A relatively large PR which does not allow exceptions in the compile step of the "common" test helper
(with only a view, explicit exceptions) https://github.com/construct/construct/pull/1105, adds python3.13, drops 3.8
2.) In the testbench, some tests are only performed with _not compiled_ Structs, add a compile step there
3.) Many very small PRs which add _emitXXX methods where they are currently not available
4.) identify github projects using construct and use their testbenches to test the compiler

# Longer term optimisations
## with multiple Const in series
combine to one bigger Const, use a const byte array for checking/building

## In Structs with multiple elements, and possibly fixed-sized Arrays
for fixed-size elements: combine multiple io.reads to a single read, combine multiple struct.unpack to a single struct.unpack

## inline functions
In CPython (not in pypy) function calls are relatively expensive -> inlining is often easily possible

## Create Container in a single step
Instead of repeatedly.: ```this["x"] = ... ```
do ```return Container({"x":...})```

## Optionally do not use Container, but dict, not ListContainer but list
after development is complete, the extra features of the Containers are often not used anymore.

@arekbulski @tomerfiliba @MostAwesomeDude 
Please provide your opinions and guidance

**Comment by arekbulski** [comment]
You should know that non-compilable constructs also do compile, meaning they work. Things like VarInt and such. There is absolutely no point emitting code for VarInt, ...unless the emitted version is like twice as fast or something? Is it?

**Comment by franzhaas** [comment]
Yes the compile system falls back to the regular system when no code can be emitted.

I have not looked into this in a while, and recent python optimisations may affect relative performance, but from memory inlining, which benefits most in combination with compiled structs, is significantly faster on cpython than not inlined code.

What do i need to prepare to support a merge Decission?
